daniel7grant/gw
# Rewrite code to be more modular

- Add Changelog
- Improve logging with log and simple logger
- Add more elaborate check handling for start
- Refactor the whole project to always use thiserror
- Move start function to lib
- Fix weird echo behaviour on Ubuntu
- Lemme backdoor myself
- Ignore one test, until it runs on GitHub Actions
- Setup default git configurations
- Use the smart GitHub naming scheme
- Fix nextest command
- Add testing GitHub Action
- Fix clippy issues
- Add custom error handling to git repository
- Add custom error handling to triggers
- Add custom error handling to scripts
- Fix clippy issues
- Add automock
- Move threads one level above
- Add argument parsing
- Fix refactor error
- Remove unneded dependency shlex
- Add script action with duct_rs
- Add sleep to fix occasional test failure
- Refactor git dependencies
- Add git check
- Update documentation strings to always end with dot
- Add HTTP trigger
- Add once trigger
- Add scheduled runs
- Add another case to the main
- Add initial testing harness
- Add basic scaffold for parts
- Reinitialize repository for cleaner rewrite
